### DISCUSSION TITLE
Add -print-settings field for Filename

### DISCUSSION THREAD

**Discussion by nallylatwit** [sumatrapdfreader/sumatrapdf#2987] [Ideas]
I have a specific action that needs to be repeated: convert many .png files to .pdf files. This is easily achieved with Microsoft's "Print to PDF" feature, using the system print dialogue. SumatraPDF has the potential to be the only readily available program capable of automating this task if the -print-settings flag included a field for the Filename. However, because there is no option to fill the Filename field in the command line, Sumatra cannot do this. If this is possible, please consider adding this as a feature.

**Comment by GitHubRulesOK** [comment]
**TL;DR see the comment below from @wqweto (Vladimir) that this feature is now added into latest pre-release**


Cant test  as no current printing but it should work without any additions, what you seem to be skipping over is that SumatraPDF "blind" prints. That is, it hands the images over to windows with the printer port details which for MS Print is Microsoft Print to PDF or xps or whatever it is then MS that redirects its output to the PortPrompt i.e it independently asks its user which port filename to use.

There are many much better routes to do what you wish more easily 

First try `mspaint /pt test.png "Microsoft Print to PDF"`
then try `write /pt pageWithImage.doc "Microsoft Print to PDF" "Microsoft Print to PDF" test.pdf`
(that last one allows you to add searchable text with image)
so Microsoft itself can work with port filenames for some apps but not others

The common work around for any application (not just SumatraPDF) is to ask an MS image to port application to print blind to file (no way to change settings) OR duplicate MS pdf printer with a known fixed filename so it has no need to ask but SumatraPDF can passthrough details like papersize (not always respected)

see here https://stackoverflow.com/a/71225924/10802527 read the points raised there then follow the link to instructions

**Comment by wqweto** [comment]
Btw, outfile in `-print-settings` is already implemented as of https://github.com/sumatrapdfreader/sumatrapdf/issues/3917
